internal/core/src/segcore/InsertRecord.h

```diff
@@ -34,8 +35,47 @@ struct InsertRecord {
     // used for timestamps index of sealed segment
     TimestampIndex timestamp_index_;
 
+    // pks to row offset
+    Pk2OffsetType pk2offset_;
+
     explicit InsertRecord(const Schema& schema, int64_t size_per_chunk);
 
+    std::vector<SegOffset>
+    search_pk(const PkType pk, Timestamp timestamp) const {
```

Comment:
Better to add comments for this method. According to the code, it will return all offsets in segment whose pk is equal to passed and timestamp is less than passed.

Reply:
And maybe this method can reuse `search_pk(const PkType pk) const`.